### ISSUE TITLE
Arch linux? What would be required for it to be supported?

### ISSUE THREAD

**Issue by Inglan** [ading2210/shimboot#130]
**Is your feature request related to a problem? Please describe.**
I use arch btw, but shimboot only supports debian.

**Describe the solution you'd like**
The ability to boot arch linux

**Describe alternatives you've considered**
This isn't a required thing, I can use debian.

**Additional context**
I have seen the note on the readme about arch

I really just posted this because I am interested in what is required. I would assume you have to patch systemd for arch linux. How hard is this? Could I help? I am not that experienced in lower level things but I think I could work it out.

**Comment by adhdkiddy** [comment]
Cant you use terraOS because that has arch support?

**Comment by Inglan** [comment]
> Cant you use terraOS because that has arch support?

didn't know abt this. looks interesting

**Comment by ading2210** [comment]
I would need to host a repository for Arch which provides the patched systemd package. I have never used Arch before so that would be rather difficult for me. The current code to build the Debian repos are located at https://github.com/ading2210/shimboot-repo, and this would have to be adapted to work with Arch as well. 

Porting [Artix Linux](https://en.wikipedia.org/wiki/Artix_Linux), which is Arch without systemd, is likely much easier since no custom repos are needed in this case. If you want to help, try writing a build script for this, in a similar manner to how the existing Alpine and Debian ones work.

**Comment by Inglan** [comment]
> I would need to host a repository for Arch which provides the patched systemd package. I have never used Arch before so that would be rather difficult for me. The current code to build the Debian repos are located at https://github.com/ading2210/shimboot-repo, and this would have to be adapted to work with Arch as well. 
> 
> Porting [Artix Linux](https://en.wikipedia.org/wiki/Artix_Linux), which is Arch without systemd, is likely much easier since no custom repos are needed in this case. If you want to help, try writing a build script for this, in a similar manner to how the existing Alpine and Debian ones work. 

Looks like terraos creator has patched systemd and published to aur (arch user repo). https://aur.archlinux.org/packages/systemd-chromiumos

I got arch working with this

**Comment by ading2210** [comment]
That AUR package isn't really ideal, since it requires systemd to be completely recompiled on every update, which might take over an hour on a typical Chromebook. Plus, afaik using the AUR for critical system packages is a bad idea since tools like pacman are unaware of AUR packages.

**Comment by Inglan** [comment]
> That AUR package isn't really ideal, since it requires systemd to be completely recompiled on every update, which might take over an hour on a typical Chromebook. Plus, afaik using the AUR for critical system packages is a bad idea since tools like pacman are unaware of AUR packages.

We could look into building packages with GitHub actions or something to solve the build times issue. 

Also I think pacman is aware of aur packages as makepkg just builds a tar.zst file which is installable through pacman.

**Comment by ading2210** [comment]
I think this should be reopened because Arch support is a meaningful and desired feature. 

> We could look into building packages with GitHub actions or something to solve the build times issue.

That's what I had in mind too. The Shimboot Debian repo already uses Github Actions for builds, and the repo itself is hosted on Cloudflare Pages (25mb file limit though). 

> Also I think pacman is aware of aur packages as makepkg just builds a tar.zst file which is installable through pacman.

Pacman cannot manage updates for AUR packages. According to the [Arch Wiki](https://wiki.archlinux.org/title/Arch_User_Repository):

> The AUR is unsupported, so any packages you install are your responsibility to update, not pacman's. If packages in the official repositories are updated, you will need to rebuild any AUR packages that depend on those libraries. 

We cannot use the AUR to host the packages. The same approach that is used for Debian, which is hosting a extra repo and telling the package manager to always prioritize it, will have to be used here.

**Comment by Inglan** [comment]
> Pacman cannot manage updates for AUR packages. According to the [Arch Wiki](https://wiki.archlinux.org/title/Arch_User_Repository):

I meant that pacman is aware of AUR packages but doesnt support them. For example it wont install systemd if I have a patched version installed

**Comment by Inglan** [comment]
Looks like you can build systemd from git with aur: https://aur.archlinux.org/packages/systemd-git

**Comment by LavedenC1** [comment]
use gnome boxes

**Comment by Inglan** [comment]
> use gnome boxes

the chromebooks already run badly just running linux on bare metal... virtualization would just make it worse.

**Comment by Inglan** [comment]
Just wrote some actions scripts that build systemd https://github.com/Inglan/systemd-chromiumos-actions-build

**Comment by Inglan** [comment]
Thanks for the reminder, added to my todo list

**Comment by Inglan** [comment]
Working on this now, I'll post my findings

**Comment by Inglan** [comment]
https://arch.mirror.constant.com/core/os/x86_64/systemd-257.2-2-x86_64.pkg.tar.zst (For example) is where systemd is located in an arch mirror

https://arch.mirror.constant.com/core/os/x86_64/core.db is some kind of database with all the packages in it. This would need to be modified. I am trying to open it now but can't for some reason...

**Comment by Inglan** [comment]
Just threw 7zip at it and got something!
![Image](https://github.com/user-attachments/assets/90ef9186-0822-4940-9b26-b8dd6af38098)

Each folder has a desc file under it

![Image](https://github.com/user-attachments/assets/a286281f-2765-44a4-9675-aee308297e79)

Here is systemd's: [desc](https://github.com/user-attachments/files/18582356/desc.txt)

It seems to have some data about conflicts, sha256 sums, packager, description, deps, etc

**Comment by Inglan** [comment]
So here is the list of what to do

- [x] Systemd patch (on the aur already: https://aur.archlinux.org/packages/systemd-chromiumos)
- [x] Actions script to build patched systemd: 
- [x] I (@Inglan) did this (https://github.com/Inglan/systemd-chromiumos-actions-build/actions)
- [ ] figure out arch linux database structure
- [ ] figure out how to make the .db file (seems to be sqlite but can't open it)
- [ ] write github actions to build + deploy
- [ ] choose hosting service to deploy to (pref cloudflare, but it has a 25mb limit)
- [ ] work out how to override systemd (separate package that conflicts and thus replaces systemd? i think this approach would be easier)
- [ ] write some build scripts using pacstrap or something with github actions

**Comment by ading2210** [comment]
It looks like you can follow https://wiki.archlinux.org/title/Pacman/Tips_and_tricks#Custom_local_repository to create a custom repo

**Comment by Inglan** [comment]
Yeah, looked into Artix and that looks good as it has basestrap to bootstrap it into a folder, so I'll try and get that working when I have the time

**Comment by Inglan** [comment]
Thanks, forgot about this. Might do it this weekend

**Comment by Inglan** [comment]
I currently don't have the time to do this, and don't have a chromebook to test it on at the moment. Alpine should be fine for now.
